src/ansys/workbench/core/workbench_launcher.py

```diff
@@ -0,0 +1,295 @@
+
+        successful = False
+        process = None
+        if self._wmi:
+            process_startup_info = self._wmi_connection.Win32_ProcessStartup.new(ShowWindow=1)
+            process_id, result = self._wmi_connection.Win32_Process.Create(
+                CommandLine=command_line, ProcessStartupInformation=process_startup_info
+            )
+            if result == 0:
+                successful = True
+                self._process_id = process_id
+                for p in self._wmi_connection.Win32_Process():
+                    if p.ProcessId == process_id:
+                        self._process = p
+                        break
+        else:
+            process = subprocess.Popen(
+                args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True
+            )
+            if process:
+                successful = True
+                self._process_id = process.pid
+                self._process = process
+        if successful:
+            logging.info("Workbench launched on the host with process id: " + str(self._process_id))
+        else:
+            logging.error("Workbench failed to launch on the host.")
+            return 0
+
+        # retrieve server port once WB is fully up running
+        port = None
+        timeout = 60 * 8  # set 8 minutes as upper limit for WB startup
+        start_time = time.time()
+        while True:
+            if self._wmi:
+                port = self.__getenv("ANSYS_FRAMEWORK_SERVER_PORT")
+            else:
+                for line in process.stdout:
+                    line = line.rstrip()
+                    if line.startswith("ANSYS_FRAMEWORK_SERVER_PORT="):
+                        port = line[28:]
+                        break
+            if port and port.startswith(prefix):
+                port = port[len(prefix) :]
+                break
+            else:
+                port = None
+            if time.time() - start_time > timeout:
+                logging.error("Failed to start Workbench service within reasonable timeout.")
+                break
+            time.sleep(10)
+        if not port or int(port) <= 0:
+            logging.error("Failed to retrieve the port used by Workbench service.")
+            return 0
+        logging.info("Workbench service uses port " + port)
+        return int(port)
+
+    def __getenv(self, key):
+        value = None
+        if self._wmi:
+            for ev in self._wmi_connection.Win32_Environment():
+                if ev.Name == key:
+                    value = ev.VariableValue
+        elif self._libc:
+            getenv = self._libc.getenv
+            getenv.restype = ctypes.c_char_p
+            value = getenv(key.encode("utf-8"))
+            if value:
+                value = value.decode("utf-8")
+        else:
+            raise Exception("unexpected code path")
+        return value
+
+    def exit(self):
+        """Terminate the launched Workbench server."""
```

Comment:
Terminate is not usually considered the best choice. Can we use "End" or "Exit" here and in the code comment in line 254?